ludwig/datasets/ohsumed/ohsumed.py

```diff
@@ -0,0 +1,93 @@
+
+
+def load(cache=None):
+    dataset = OhsuMed(cache)
+    return dataset.load()
+
+
+class OhsuMed(BaseDataset):
+
+    def __init__(self, passed_in_cache_dir):
+        super().__init__("ohsumed")
+        if passed_in_cache_dir is not None:
+            self._cache_dir = passed_in_cache_dir
+        self._source_location = os.path.join(self._initial_path, "./text/ohsumed/ohsumed-allcats.csv")
+        self._cur_version = self._config_file_contents["ohsumed"]
+
+    """Download the ohsumed raw data to the ludwig cache in the format ~/.ludwig_cache/id
+       where is is represented by the name.version of the dataset
+       :param dataset_name: (str) the name of the dataset we need to retrieve.
+       Returns: 
+          None
+    """
+    def download(self, dataset_name) -> None:
+        download_path = Path.home().joinpath('.ludwig_cache').joinpath("ohsumed_"
+                                                       + str(self._cur_version)).joinpath('raw.csv')
+        shutil.copy(self._source_location, download_path)
+        result = os.path.isfile(download_path)
+        if not result:
+            raise FileNotFoundError("The raw data was not downloaded correctly")
+
+    """Process the ohsumed dataset to get it ready to be plugged into a dataframe
+       in the manner needed by the ludwig training API, to do this we create
+       a new dictionary that contains the KV pairs in the format that we need.
+       Returns:
+           None
+    """
+    def process(self) -> None:
+        if super().check_file_existence(self._cur_version, "ohsumed", "raw.csv"):
+            dict_reader = csv.DictReader(open(self._download_path))
+            value_to_store = None
+            for row in dict_reader:
+                for key, value in row.items():
+                    if key == "class":
+                        value_to_store = value
+                    else:
+                        key_to_store = value
+                        self._result_dict[key_to_store] = value_to_store
+        else:
+            raise FileNotFoundError("The raw dataset for ohsumed was not found")
+        try:
+            with open(self.processed_data_path, 'w') as csv_file:
+                writer = csv.writer(csv_file)
+                for key, value in self._result_dict.items():
+                    writer.writerow([key, value])
+        except IOError:
+            print("I/O error")
+
+    """Now that the ohsumed data is processed load and return it as a pandas dataframe
+       Returns:
+          A pandas DataFrame
+    """
+    def load(self) -> pd.DataFrame:
+        column_names = ["text", "class"]
+        if super().check_file_existence(self._cur_version, "ohsumed", "processed.csv"):
+            return pd.read_csv(self.processed_data_path, names=column_names)
+        else:
+            raise FileNotFoundError("The transformed data for ohsumed does not exist so cant return pandas dataframe")
```

Comment:
Instead or raising an exception, we can just call `self.preprocess()` first.

Reply:
We redo the previous steps now

Reply:
Again check the mixin logic